src/System.Buffers.Experimental/System/Range.cs

```diff
@@ -1,26 +1,82 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-namespace System {
-    public struct Range
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System
+{
+
+    public struct Range : IEnumerable<int>
     {
-        public Range(long index, long length)
+        public const int UnboundedFirst = Int32.MinValue;
+        public const long UnboundedLast = Int32.MaxValue;
+
+        public readonly int First;
+        public readonly int Length;
+        public long Last => ((long)Length) + First - 1;
+
+        public Range(int first, int length)
         {
             if (length < 0)
             {
                 throw new ArgumentOutOfRangeException("Length must be non-negative.", nameof(length));
             }
 
-            Index = index;
+            First = first;
             Length = length;
         }
-        public readonly long Index;
-        public readonly long Length;
 
-        public void Deconstruct(out long index, out long length)
+        public void Deconstruct(out int first, out long last)
+        {
+            first = First;
+            last = Last;
+        }
+        public static Range Construct(int first, long last = UnboundedLast)
+        {
+            if (last == UnboundedLast) return new Range(first, int.MaxValue);
```

Comment:
Should this be `last >= UnboundedLast`?

Reply:
How can last be >= long.MaxValue?

Reply:
BTW, I thought about this whole design more. I want to remove any longs from the surface area. I updated the PR